drivers/mipi_dbi/mipi_dbi_smartbond.c

```diff
@@ -0,0 +1,610 @@
+		if (ret < 0) {
+			LOG_WRN("Could not switch to PLL. Requested speed should not be achieved.");
+		}
+	}
+#endif
+
+	return mipi_dbi_smartbond_configure(dev);
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int mipi_dbi_smartbond_suspend(const struct device *dev)
+{
+	const struct mipi_dbi_smartbond_config *config = dev->config;
+	int ret;
+
+	/* Select sleep state; it's OK if settings fails for any reason. */
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+	if (ret < 0) {
+		LOG_WRN("Could not apply MIPI DBI pins' sleep state. Consumption might be increased.");
+	}
+
+	/* Disable host controller to minimize power consumption. */
+	da1469x_lcdc_set_status(false, false, 0);
+
+	return 0;
+}
+
+static int mipi_dbi_smartbond_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		ret = mipi_dbi_smartbond_suspend(dev);
+
+#if CONFIG_PM_DEVICE_RUNTIME && CONFIG_PM
+		da1469x_pd_release(MCU_PD_DOMAIN_SYS);
+#endif
+	case PM_DEVICE_ACTION_RESUME:
+		ret = mipi_dbi_smartbond_resume(dev);
+
+#if CONFIG_PM_DEVICE_RUNTIME && CONFIG_PM
+		/*
+		 * If resume succeeded, prevent the system from entering the
+		 * extended sleep state (Instead, use __WFI).
+		 */
+		if (ret == 0) {
+			da1469x_pd_acquire(MCU_PD_DOMAIN_SYS);
+		}
+#endif
+	default:
+		return -ENOTSUP;
+	}
+
+	return ret;
+}
+#endif
+
+static int mipi_dbi_smartbond_init(const struct device *dev)
+{
+	const struct mipi_dbi_smartbond_config *config = dev->config;
+	struct mipi_dbi_smartbond_data *data = dev->data;
+	int ret;
+
+	/* Device should be ready to be acquired */
+	k_sem_init(&data->device_sem, 1, 1);
+	/* Event should be signaled by LCDC ISR */
+	k_sem_init(&data->sync_sem, 0, 1);
+
+#if MIPI_DBI_SMARTBOND_IS_RESET_REQUIRED
+	if (config->reset.port) {
+		 ret = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_INACTIVE);
+		 if (ret < 0) {
+			LOG_ERR("COuld not configure reset line (%d)", ret);
+			return -EIO;
+		 }
+	}
+#endif
+
+	IRQ_CONNECT(SMARTBOND_IRQN, SMARTBOND_IRQ_PRIO, smartbond_mipi_dbi_isr,
+													DEVICE_DT_INST_GET(0), 0);
+
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+	/* Make sure device state is marked as suspended */
+	pm_device_init_suspended(dev);
+
+	ret = pm_device_runtime_enable(dev);
+	if ((ret < 0) && (ret != -ENOSYS)) {
+		return ret;
+	}
+#else
+	/* Resme if either PM is not used at all or if PM without runtime is used. */
+	ret = mipi_dbi_smartbond_resume(dev);
+	if (ret < 0) {
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+static struct mipi_dbi_driver_api mipi_dbi_smartbond_driver_api = {
+#if MIPI_DBI_SMARTBOND_IS_RESET_REQUIRED
+	.reset = mipi_dbi_smnartbond_reset,
+#endif
+	.command_write = mipi_dbi_smartbond_command_write,
+	.write_display = mipi_dbi_smartbond_write_display,
+#if MIPI_DBI_SMARTBOND_IS_READ_SUPPORTED
+	.command_read = mipi_dbi_smartbond_command_read,
+#endif
+};
+
+#define SMARTBOND_MIPI_DBI_INIT(inst)	\
```

Comment:
I mentioned this above- but we should make this driver support multiple instances. The macros here are already instance based, so if there is a compelling reason to keep the driver single instance we need to use single instance macros here as well.

Reply:
From architecture point of view, our SoC integrates a single LCDC IP block. The only reason that multiple instances should be supported is to cover use cases where more than one display device is to be driven. Note sure if this scenario has a meaning in a real application. @danieldegrasse what do you think?